core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -109,6 +118,7 @@ class ReplicaManager(val config: KafkaConfig,
                      val logManager: LogManager,
                      val isShuttingDown: AtomicBoolean,
                      quotaManager: ReplicationQuotaManager,
+                     val metadataCache: MetadataCache,
```

Comment:
Sure. I have updated the code to remove `metadataCache` from parameters of methods `becomeLeaderOrFollower`, `maybeUpdateMetadataCache` and `makeFollowers`.